kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/asset/Asset.java

```diff
@@ -114,10 +112,10 @@
      * @throws NullPointerException
      *             if any of the arguments is null
      * @throws IllegalArgumentException
-     *             channel ID is less than or
-     *             equal to zero
+     *             If the provided channel name is no present in the configuration
+     *             of this asset
      */
-    public void registerAssetListener(long channelId, AssetListener assetListener) throws KuraException;
+    public void registerAssetListener(String channelName, AssetListener assetListener) throws KuraException;
```

Comment:
Shouldn't it be better to check if a channel is invalid first and if not throw `IllegalArgumentException`. Currently according to the contract, it will try to find the provided channel in the Asset Channel Configuration and if not found, it will throw `IllegalArgumentException`. In either way, it seems same to me but I believe from a good contract point of view we must check if the channel name is invalid. What do you think?